docs/contributing/test-data-models-overview.md

```diff
@@ -0,0 +1,250 @@
+  .build();
+```
+
+#### Building predefined version data models
+
+```ts
+import {
+  RestModelBuilder,
+  GraphqlModelBuilder,
+  type TGeometryRest,
+  type TGeometryGraphql,
+} from '@commercetools-test-data/geometry';
+
+const restGeometry: TGeometryRest = RestModelBuilder.presets.munich().build();
+const graphqlGeometry: TGeometryGraphql = GraphqlModelBuilder.presets
+  .munich()
+  .build();
+```
+
+## Internal structure
+
+The way a data model is implemented is by using some classes and helpers exposed from the [core](https://github.com/commercetools/test-data/tree/main/core) package.
+
+### Fields config
+
+The core file is named `fields-config` and its responsibility is to describe how the data model properties should be pre-populated and also resolve dependencies among them (if any).
+
+Here's an example of a simple one:
+
+```ts
+import { fake, type TModelFieldsConfig } from '@commercetools-test-data/core';
+import type { TGeometryGraphql, TGeometryRest } from './types';
+
+const commonFieldsConfig = {
+  type: fake(() => 'Point'),
+  coordinates: fake((f) => [f.location.longitude(), f.location.latitude()]),
+};
+
+export const restFieldsConfig: TModelFieldsConfig<TGeometryRest> = {
+  fields: {
+    ...commonFieldsConfig,
+  },
+};
+
+export const graphqlFieldsConfig: TModelFieldsConfig<TGeometryGraphql> = {
+  fields: {
+    ...commonFieldsConfig,
+    __typename: 'Geometry',
+  },
+};
+```
+
+As you can see, this files is required to export two configuration objects (`restFieldsConfig`, `graphqlFieldsConfig`) which describe how to populate the properties of the data model when building the default version.
+
+The configuration objects has two properties:
+
+- `fields` (**required**): Describe how the default values should be populated
+- `postBuild` (_optional_): Callback function to run modifications in the built data model
+
+Here's an example where we use `postBuild` option to populate some values which depend on others to be built first:
+
+```ts
+export const graphqlFieldsConfig: TModelFieldsConfig<TChannelGraphql> = {
+  fields: {
+    ...commonFieldsConfig,
+    __typename: 'Channel',
+    nameAllLocales: null,
+    descriptionAllLocales: null,
+  },
+  postBuild: (model) => {
+    const name = model.nameAllLocales
+      ? LocalizedString.resolveGraphqlDefaultLocaleValue(model.nameAllLocales)
+      : undefined;
+    const description = model.descriptionAllLocales
+      ? LocalizedString.resolveGraphqlDefaultLocaleValue(
+          model.descriptionAllLocales
+        )
+      : undefined;
+    return {
+      name,
+      description,
+    };
+  },
+};
+```
+
+In this example, `name` and `description` are values that are calculated based on the `nameAllLocales` and `descriptionAllLocales` ones but we need the latter to be built first, so that why we use the `postBuild` function.
+
+This function just need to return an object with the updated properties values and it will be merged with the originally built one.
```

Comment:
```suggestion
This function just need to return an object with the updated properties and their values and it will be merged with the originally built one.
```

Reply:
Updated here: 620a0125473afc82bd57cb988907d7ff8e8aa740